indico/modules/announcement/templates/display.html

```diff
@@ -1,23 +1,80 @@
-<div class="announcement-bar weak-hidden" id="announcement-bar">
-    <div class="flexrow f-self-stretch f-j-center f-a-center">
-        <i class="icon icon-warning"></i>
-        <div>{{- message | safe -}}</div>
-    </div>
-    <span class="icon-close close-button"></span>
-</div>
-<script>
-    (function() {
-        'use strict';
+{% macro render_announcement_legacy(type, message, persistent=false) %}
+    {% if type == 'html' %}
+        {{ message | safe }}
+    {% else %}
+        <div class="announcement-bar weak-hidden {{type}}" data-hash={{message | crc32 | string | tojson}}>
+            <div class="flexrow f-self-stretch f-j-center f-a-center">
+                <i class="icon icon-warning"></i>
+                <div>
+                    {{ message }}
+                </div>
+            </div>
+            {% if not persistent %}
+                <span class="icon-close close-button"></span>
+            {% endif %}
+        </div>
+    {% endif %}
+{% endmacro %}
 
-        var hash = {{ message | crc32 | string | tojson }};
+{% macro render_announcement_new(type, message, persistent=false) %}
+    {% if type == 'html' %}
+        {{ message | safe }}
+    {% elif type == 'logged-in-as' %}
```

Comment:
Maybe removing this line and replacing `else` with `elif type != 'logged-in-as' %}` would look nicer?

Reply:
I actually forgot that line there. It should be removed, since this case is handled elsewhere.
Good catch! 👍